routers/admin.py

```diff
@@ -470,6 +485,117 @@ def get_filtered_user_query(
     return user_query.get_complete_query(admin_roles, pagination)
 
 
+def _count_users_with_membership_status(
+    *,
+    db_session: Session,
+    admin_roles: list[str],
+    base_params: dict[str, object],
+    status: ApprovalStatusEnum,
+) -> int:
+    """
+    Count distinct users who have either a platform or group membership
+    with the given status, limited to resources the admin can manage.
+    """
+    # Remove status filters so we can apply OR logic below
+    params_data = {**base_params, "platform_approval_status": None, "group_approval_status": None}
+    params = UserQueryParams(**params_data)
+    allowed_platforms_subquery, allowed_groups_subquery = _get_allowed_resource_subqueries(admin_roles)
+
+    base_conditions = [
+        params.get_admin_permissions_query(admin_roles),
+        *params.get_query_conditions(),
+    ]
+
+    platform_status_query = select(PlatformMembership.user_id).where(
```

Comment:
Should this logic be used in `UserQueryParams.platform_approval_status_query`? It probably makes sense that we only consider platforms the admin has access to. That way we wouldn't have to have separate logic here.